### PR TITLE
Mac py3

### LINKED ISSUE #13628: Support python3 build and test in Mac OSX
According to the requirements [here](https://github.com/PaddlePaddle/Paddle/issues/12040), we should support python3 in mac

### PR DESCRIPTION
**Pull request by velconia**
This close #13628

### CODE REVIEW

File: paddle/scripts/paddle_build.sh [hunk 1]
```diff
@@ -82,7 +82,18 @@ function cmake_gen() {
             else
                 exit 1
             fi
-        # TODO: qiyang add python3 part here 
+        elif [ "$1" == "cp35-cp35m" ]; then
+            if [ -d "/Library/Frameworks/Python.framework/Versions/2.7" ]; then
```
Comment (JiabinYang) [inline paddle/scripts/paddle_build.sh]: This part should be py35

**Review by JiabinYang — CHANGES_REQUESTED** [review]

File: paddle/scripts/paddle_build.sh (hunk 1 above)
Comment (velconia) [reply paddle/scripts/paddle_build.sh]: Thanks, I fixed it now

File: paddle/scripts/paddle_build.sh [hunk 2]
```diff
@@ -82,7 +82,18 @@ function cmake_gen() {
...
-        # TODO: qiyang add python3 part here 
+        elif [ "$1" == "cp35-cp35m" ]; then
+            if [ -d "/Library/Frameworks/Python.framework/Versions/3.5" ]; then
+                export LD_LIBRARY_PATH=/Library/Frameworks/Python.framework/Versions/3.5/lib/
+                export DYLD_LIBRARY_PATH=/Library/Frameworks/Python.framework/Versions/3.5/lib/
+                export PATH=/Library/Frameworks/Python.framework/Versions/3.5/bin/:${PATH}
+                PYTHON_FLAGS="-DPYTHON_EXECUTABLE:FILEPATH=/Library/Frameworks/Python.framework/Versions/3.5/bin/python3
+            -DPYTHON_INCLUDE_DIR:PATH=/Library/Frameworks/Python.framework/Versions/3.5/include/python3.5m/
+            -DPYTHON_LIBRARY:FILEPATH=/Library/Frameworks/Python.framework/Versions/3.5/lib/libpython3.5m.dylib"
+                WITH_FLUID_ONLY=${WITH_FLUID_ONLY:-ON}
```
Comment (JiabinYang) [inline paddle/scripts/paddle_build.sh]: Should not open FLUID_ONLY

**Review by JiabinYang — APPROVED** [review]
LGTM

**Pull request merged**